nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java

```diff
@@ -120,22 +132,35 @@ public String export(CoreSession session, String nxql,
         } catch (IOException e) {
             throw new NuxeoException("Unable to process stats blob", e);
         }
-        List<Map<String, String>> inputs = propsToTypedList(inputProperties);
-        List<Map<String, String>> outputs = propsToTypedList(outputProperties);
-        List<Map<String, String>> featuresWithType = new ArrayList<>(inputs);
+        List<IOParam> inputs = propsToTypedList(inputProperties);
+        List<IOParam> outputs = propsToTypedList(outputProperties);
+
+        List<IOParam> featuresWithType = new ArrayList<>(inputs);
         featuresWithType.addAll(outputs);
-        DocumentModel corpus = createCorpus(session, nxql, inputs, outputs, split, statsBlob);
+        AICorpus corpus = createCorpus(session, nxql, inputs, outputs, split, statsBlob);
 
         List<String> featuresList = new ArrayList<>(inputProperties);
         featuresList.addAll(outputProperties);
-        BulkCommand bulkCommand = new BulkCommand.Builder(EXPORT_ACTION_NAME, notNullNxql(nxql, featuresWithType))
+
+        String notNullNXQL = notNullNxql(nxql, featuresWithType);
+        String username = session.getPrincipal().getName();
+        BulkCommand bulkCommand = new BulkCommand.Builder(EXPORT_ACTION_NAME, notNullNXQL, username)
```

Comment:
Need to check whether this constructor is available on 10.10

Reply:
it's not but the previous one was deprecated.

Reply:
we must support 10.10

Reply:
not arguing with that. It's master branch though. Aligned on non LTS Nuxeo and we use APIs that will be deprecated at some point. Just wanted to avoid that. Moving this part to 10.10 will requires some changes but at some point this difference will be there. So why not do it now?

Reply:
AI is an addon so it doesn't necessaryily follow the versioning of the platform.  I am trying to keep our `master` branch compatible with 10.10

Reply:
moved back to deprecated constructor